include/geode/mesh/builder/tetrahedral_solid_view_builder.h

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2019 - 2020 Geode-solutions
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#pragma once
+
+#include <vector>
+
+#include <geode/mesh/builder/tetrahedral_solid_builder.h>
+#include <geode/mesh/common.h>
+
+namespace geode
+{
+    FORWARD_DECLARATION_DIMENSION_CLASS( Point );
+    FORWARD_DECLARATION_DIMENSION_CLASS( TetrahedralSolidView );
+} // namespace geode
+
+namespace geode
+{
+    /*!
+     * Implementation class for TetrahedralSolidBuilder using
+     * TetrahedralSolidView data structure
+     */
+    template < index_t dimension >
+    class TetrahedralSolidViewBuilder
+        : public TetrahedralSolidBuilder< dimension >
+    {
+        OPENGEODE_TEMPLATE_ASSERT_3D( dimension );
+
+    public:
+        static std::unique_ptr< TetrahedralSolidViewBuilder< dimension > >
+            create( TetrahedralSolidView< dimension >& mesh );
+
+        void add_viewed_vertex( index_t vertex_id );
```

Comment:
I have not thinking of it before but what do you think about method like `add_viewed_vertices(absl::Span<const index_t> vertex_ids)` and `add_viewed_tetrahedra(absl::Span<const index_t> tetra_ids)`

Reply:
maybe. It will not change the way it works behind the scene (adding one at the time)

Reply:
Yes, you are right. The only question is where should it be done : in the class or in the client codes. I propose to keep that in mind but not apply changes right now (let's see how it is used in practice)